cpp/daal/src/services/service_defines.h

```diff
@@ -42,7 +41,11 @@ DAAL_EXPORT bool daal_check_is_intel_cpu();
     #define DAAL_BASE_CPU daal::rv64
 #endif
 
-#define DAAL_CHECK_CPU_ENVIRONMENT (daal_check_is_intel_cpu())
+#if defined(TARGET_X86_64)
+    #define DAAL_CHECK_CPU_ENVIRONMENT 1
+#else
+    #define DAAL_CHECK_CPU_ENVIRONMENT 0
+#endif
```

Comment:
Please revert this change as daal_check_is_intel_cpu() can return different values on x86.

The code of the library contains branches that were not tested on non-Intel x86 CPUs with this change and thus might produce incorrect or suboptimal results.

Reply:
@isuruf Can you please address this request?

Reply:
@isuruf No, reverting this change will only leave the throttling to sse2 in `DAAL_DISPATCH_FUNCTION_BY_CPU_SAFE` macros which are used in only couple of the algorithms.
You can remove them and replace with `DAAL_DISPATCH_FUNCTION_BY_CPU` to prevent the throttling in those algorithms.

We do not run regular testing on AMD machines and our code contains branches that depend on this `DAAL_CHECK_CPU_ENVIRONMENT` macro. I.e. the logic of the code will change on AMD after setting this macro to 1. This could potentially lead to errors in the computations on AMD.

Reply:
> What kind of testing would you require for this to be included? The PR I sent was to avoid throttling to sse2 on AMD and reverting this change would make this PR not useful.

Just to expand answer on this. 
1. it's ok to go with cpuid portion and do dispatching on ISA level.
2. But we still need vendor specialization for some of optimizations - like cache handling or numa handling because Intel and AMD cpus have different hierarchy structures.  so this portion have to be preserved, but now they both done by same switch

Reply:
Okay, let's get this PR with just the CPUID in and leave the rest for another PR/issue.

Reply:
@Vika-F have an idea how to rearrange second portion and would followup

Reply:
@Vika-F @napetrov any updates on this? I can help if you need it.